README.md

```diff
@@ -917,39 +921,91 @@ end
+
+At the first run it will reset everything because it needs to lock all the index specifications in the `Chewy::Stash`.
+
+If it is run twice in a row, the second time nothing will be reset.
+
+See [Chewy::Stash](lib/chewy/stash.rb) and [Chewy::Index::Specification](lib/chewy/index/specification.rb) for more details.
 
+
+```bash
+rake chewy:upgrade # upgrades all the existing indices, declared in app/chewy
+rake chewy:upgrade[users] # upgrades UsersIndex only
+rake chewy:upgrade[users,places] # upgrades UsersIndex and PlacesIndex
+rake chewy:upgrade[-users,places] # upgrades every index in application except specified ones
+```
+
+#### `chewy:update`
+
+Doesn't create indexes, it simply imports everything to the existing ones and fails if the index was not created before.
+
+Unlike `reset` or `upgrade` tasks, it is possible to pass type references to update the particular type. In index name is passed without the type specified, it will update all the types defined for this index.
+
+```bash
 rake chewy:update # updates all the existing indices, declared in app/chewy
 rake chewy:update[users] # updates UsersIndex only
-rake chewy:update[users,projects] # updates UsersIndex and ProjectsIndex
-rake chewy:update[-users,projects] # updates every index in application except specified ones
-
+rake chewy:update[users,places#city] # updates the whole UsersIndex and PlacesIndex::City type
+rake chewy:update[-users,places#city] # updates every index in application except all the types defined in UsersIndex and the rest of the types defined in PlacesIndex
 ```
 
-`rake chewy:reset` performs zero-downtime reindexing as described [here](https://www.elastic.co/blog/changing-mapping-with-zero-downtime). So basically rake task creates a new index with uniq suffix and then simply aliases it to the common index name. The previous index is deleted afterwards (see `Chewy::Index.reset!` for more details).
+#### `chewy:sync`
 
-#### `chewy:deploy`
+Provides a way to synchronize outdated indexes with the source quickly and without doing a full reset.
 
-This rake task is especially useful during the production deploy. Currently it executes selective reset, this means that an index will be reset only if the index specification (settings or mappings) has been changed, otherwise the reset of this index will be skipped.
+Arguments are similar to the `chewy:update` ones. It is possible to specify a particular type or a whole index.
 
-Obviously at the first run it will reset everything because it needs to lock all the index specifications in the `Chewy::Stash`.
+See [Chewy::Type::Syncer](lib/chewy/type/syncer.rb) for more details.
 
-See [Chewy::Stash](lib/chewy/stash.rb) and [Chewy::Index::Specification](lib/chewy/index/specification.rb) for more details.
+```bash
+rake chewy:sync # synchronizes all the existing indices, declared in app/chewy
```

Comment:
The comment is not exactly true. It will sync all indexes that inherit from Chewy::Type. You can setup different chewy index definition location.